standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -4220,6 +4236,24 @@ private Set<MColumnDescriptor> detachCdsFromSdsNoTxn(
     }
   }
 
+  private Set<MSerDeInfo> detachSerDeInfosFromSdsNoTxn(String catName, String dbName, String tblName,
```

Comment:
could this method be unified with detachCdsFromSdsNoTxn? as well as removeUnusedColumnDescriptor and removeUnusedSerDeInfo?